File: JavaScript(NodeJS)/new_approach/new_approach.js
const readline = require('readline');

function reverse_num(number)
{
	let arr = number.split('');
	let len = arr.length;
	let bound = parseInt(len / 2);
	let i = 0;
	let symmetry = len - 1 - i;
	let temp = '';

	for (i = 0; i < bound; i++)
	{
		symmetry = len - 1 - i;
		temp = arr[i];
		arr[i] = arr[symmetry];
		arr[symmetry] = temp;
	}

	return arr.join('');
}

const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

rl.question('Please insert a number: ', (answer) => {
	let res = reverse_num(answer);
	console.log(res);

	rl.close();
});

File: Go/new_approach/new_approach.go
package main

import (
	"fmt"
)

type Reverse struct {
	number string
}

func(rev Reverse) ReverseNumber() string {
	var len int = len(rev.number)
	arr := []rune(rev.number)

	for i := 0; i < len / 2; i++ {
		arr[i], arr[len - 1 - i] = arr[len - 1 - i], arr[i]
	}
	
	return string(arr)
}

func main() {
	var input string
	fmt.Scanln(&input)
	
	rev := Reverse{number: input}
	var res string = rev.ReverseNumber()
	fmt.Println(res)
}